tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -116,6 +118,29 @@ public function storeProductReferenceByName(string $productReference, string $pr
         $this->getSharedStorage()->set($productReference, $productId);
     }
 
+    /**
+     * @Then product ":productReference" should not be associated to shops ":shopReferences"
+     *
+     * @param string $productReference
+     * @param string $shopReferences
+     */
+    public function assertProductIsNotAssociatedToShops(string $productReference, string $shopReferences): void
```

Comment:
There is already a step that does that: https://github.com/PrestaShop/PrestaShop/blob/523f87ea6e156c06d1e1db6efc3719586dbe94ed/tests/Integration/Behaviour/Features/Context/Domain/Product/ProductShopFeatureContext.php#L46

It only accepts a single reference for now, but you can modify it to accept a list instead of creating this new step (you can define to sentences on the same end point to handle the plural wording)